app-backend/db/src/main/scala/ObjectPermissions.scala

```diff
@@ -295,4 +295,21 @@ trait ObjectPermissions[Model] {
       case false =>
         queryObjectsF(user, objectType, actionType)
     }
+
+  def isReplaceWithinScopedLimit(
+      domain: Domain,
+      user: User,
+      acrList: List[ObjectAccessControlRule]
+  ): Boolean =
+    (Scopes.resolveFor(
+      domain,
+      Action.Share,
+      user.scope.actions
+    ) map { action =>
+      (action.limit map { limit =>
+        (acrList map { acr =>
+          acr.subjectId.getOrElse("")
+        }).distinct.size.toLong == limit
+      }) getOrElse { true }
+    }) getOrElse { false }
```

Comment:
This is a bit tough to read. If I'm reading it correctly, then in words, it's something like:

- get the limit
- if the limit is present, check whether the count of distinct subject ids + things without subject ids is equal to the limit
- otherwise it's unlimited so it's fine
- otherwise they don't have this scope so it's not fine

I think it's the case that we don't want the check to check equality -- but `<=` the limit, right?

And that the inner comparison might be more legible with some comments as:

```suggestion
      (action.limit map { limit =>
        val distinctUsers = acrList.foldLeft(Set.empty[String])({
          case (accum: Set[String], ObjectAccessControlRule(SubjectType.User, Some(subjId), _)) => accum | Set(subjId)
          case (accum: Set[String], _) => accum
        })
        distinctUsers.size.toLong <= limit
      }) getOrElse { true } // if there's no limit, then the limit is infinity, so this action is allowed
    }) getOrElse { false } // if there's no scope, then the action is not allowed
```

since `Set`s already express distinctness and there's no ad hoc handling of missing user ids via `getOrElse("")`.

I'm also curious about the `getOrElse("")` -- that's for cases when something is shared with everyone right? Do we have an expected behavior in that case? also, is the sharing limit then expressing _entities_ you can share with, whether those are teams or orgs or platforms or everyone-considered-as-one-entity? I hadn't considered this angle and I'm curious to hear your thoughts about it.